EventFilter/L1TXRawToDigi/plugins/L1TTwinMuxRawToDigi.cc

```diff
@@ -369,347 +380,542 @@ void L1TTwinMuxRawToDigi::processFed( int twinMuxFed,
+
+			int mb3_eta_HQ = ( dataWordSub >> 8  ) & 0xFF;  // positions  8 -> 15
+			int mb2_eta_HQ = ( dataWordSub >> 32 ) & 0xFF;  // positions 32 -> 39
+			int mb1_eta_HQ = ( dataWordSub >> 48 ) & 0xFF;  // positions 48 -> 55
+
+			if ( debug_ ) logfile << '[' << ++lcounter << "]\t" << std::hex
+								  << dataWordSub << std::dec << "\t|\t"
+								  << "mb1_eta " << mb1_eta  << '\t'
+								  << "mb2_eta " << mb2_eta  << '\t'
+								  << "mb3_eta " << mb3_eta  << '\t'
+								  << "mb1_eta_HQ " <<  mb1_eta_HQ   << '\t'
+								  << "mb2_eta_HQ " <<  mb2_eta_HQ   << '\t'
+								  << "mb3_eta_HQ " <<  mb3_eta_HQ   << '\n';
+			
+			//MB1
+			posBTI[0] = (mb1_eta & 0x01);
+			posBTI[1] = ((mb1_eta & 0x02)>>1);
+			posBTI[2] = ((mb1_eta & 0x04)>>2);
+			posBTI[3] = ((mb1_eta & 0x08)>>3);
+			posBTI[4] = ((mb1_eta & 0x10)>>4);
+			posBTI[5] = ((mb1_eta & 0x20)>>5);
+			posBTI[6] = (((mb1_eta & 0x40)>>6) || ((mb1_eta & 0x80)>>7));
+
+			qualBTI[0] = (mb1_eta_HQ & 0x01);
+			qualBTI[1] = ((mb1_eta_HQ & 0x02)>>1);
+			qualBTI[2] = ((mb1_eta_HQ & 0x04)>>2);
+			qualBTI[3] = ((mb1_eta_HQ & 0x08)>>3);
+			qualBTI[4] = ((mb1_eta_HQ & 0x10)>>4);
+			qualBTI[5] = ((mb1_eta_HQ & 0x20)>>5);
+			qualBTI[6] = (((mb1_eta_HQ & 0x40)>>6) || ((mb1_eta_HQ & 0x80)>>7));
+
+			theSegments.push_back( L1MuDTChambThDigi( bxNr, wheel, sector-1, 1, posBTI, qualBTI) );
+			
+			//MB2
+			posBTI[0] = (mb2_eta & 0x01);
+			posBTI[1] = ((mb2_eta & 0x02)>>1);
+			posBTI[2] = ((mb2_eta & 0x04)>>2);
+			posBTI[3] = ((mb2_eta & 0x08)>>3);
+			posBTI[4] = ((mb2_eta & 0x10)>>4);
+			posBTI[5] = ((mb2_eta & 0x20)>>5);
+			posBTI[6] = (((mb2_eta & 0x40)>>6) || ((mb2_eta & 0x80)>>7));
+
+			qualBTI[0] = (mb2_eta_HQ & 0x01);
+			qualBTI[1] = ((mb2_eta_HQ & 0x02)>>1);
+			qualBTI[2] = ((mb2_eta_HQ & 0x04)>>2);
+			qualBTI[3] = ((mb2_eta_HQ & 0x08)>>3);
+			qualBTI[4] = ((mb2_eta_HQ & 0x10)>>4);
+			qualBTI[5] = ((mb2_eta_HQ & 0x20)>>5);
+			qualBTI[6] = (((mb2_eta_HQ & 0x40)>>6) || ((mb2_eta_HQ & 0x80)>>7));
+
+			theSegments.push_back( L1MuDTChambThDigi( bxNr, wheel, sector-1, 2, posBTI, qualBTI) );
+			
+			//MB3
+			posBTI[0] = (mb3_eta & 0x01);
+			posBTI[1] = ((mb3_eta & 0x02)>>1);
+			posBTI[2] = ((mb3_eta & 0x04)>>2);
+			posBTI[3] = ((mb3_eta & 0x08)>>3);
+			posBTI[4] = ((mb3_eta & 0x10)>>4);
+			posBTI[5] = ((mb3_eta & 0x20)>>5);
+			posBTI[6] = (((mb3_eta & 0x40)>>6) || ((mb3_eta & 0x80)>>7));
+
+			qualBTI[0] = (mb3_eta_HQ & 0x01);
+			qualBTI[1] = ((mb3_eta_HQ & 0x02)>>1);
+			qualBTI[2] = ((mb3_eta_HQ & 0x04)>>2);
+			qualBTI[3] = ((mb3_eta_HQ & 0x08)>>3);
+			qualBTI[4] = ((mb3_eta_HQ & 0x10)>>4);
+			qualBTI[5] = ((mb3_eta_HQ & 0x20)>>5);
+			qualBTI[6] = (((mb3_eta_HQ & 0x40)>>6) || ((mb3_eta_HQ & 0x80)>>7));
+
+			theSegments.push_back( L1MuDTChambThDigi( bxNr, wheel, sector-1, 3, posBTI, qualBTI) );
+			
+		  }//etha word
+		 
+		  else if ( selector == 0xB ) { //MB1/2 output word
+	 
+			int mb2_phi =    ( dataWordSub & 0xFFF);        // positions  0 -> 11
```

Comment:
please consider a helper function to avoid duplication below

Reply:
@davidlange6 The code your are referring to is the original part that we did not want to touch at all. It was just put in the second part of the switch differentiating between the old data and ZS data. It appears to be modified due to the corrected indentation. 
I would prefer not to touch it, as we tried to minimize changes to the (old working) code.

Reply:
its true the github diffs look to be lost. I'll try again to compare...